### DISCUSSION TITLE
Ovlaim Ceiling Fan and Light identified as Casafan

### DISCUSSION THREAD

**Discussion by lk229** [make-all/tuya-local#1728] [Q&A] [answered]
I have two Ovlaim ceiling fans and light fixtures that match the product id in the ovlaim_ceiling_fan_light.yaml device file. However, on installation the only choice I have for a matching item is the casafan. Because of this, I don't get access to the led color temperature. 

I've copied the appropriate dp (17) from the ovlaim device file to the casafan device file and then my fan lighting works correctly (I also cleared the _pycache and re-configured each device but don't know if that was necessary). 
Here's my product info from the Tuya api...

    "model": "FP9805_TAIDE",
    "name": "Ceiling Fan With Light",
    "product_id": "g0ewlb1vmwqljzji",
    "product_name": "NEW_FP智能风扇灯_COMBO",

This is a screenshot from my install...
![2024-03-03_crop](https://github.com/make-all/tuya-local/assets/36109757/242f1ba8-9f9a-4180-8ea5-367aaa8a81b3)

Any tips on how to get my devices recognized correctly with modifying files that get overwritten upon updates?

**Comment by make-all** [comment]
Issue #749 that originally requested support for this device did not come with logs, so most likely the config is incorrect.
If you attach the device diagnostics, I will be able to confirm. It looks like the only difference affecting device detection is that the fan speed is integer for Casafan, but string for Ovlaim.

**Reply by lk229** [reply]
This is info from the Download Diagnostics on your Tuya Local, devices entry...
[ovlaim_diagnostics.txt](https://github.com/make-all/tuya-local/files/14548269/ovlaim_diagnostics.txt)

This second file is info retrieved from the tuya developer site...
[ovlaim_tuya_info.txt](https://github.com/make-all/tuya-local/files/14548277/ovlaim_tuya_info.txt)

**Reply by make-all** [reply]
Thanks, that confirms that the speed is indeed an unquoted integer, contrary to the original report, and also that the timer values are all different than what was originally reported.

**Reply by lk229** [reply]
I don't understand why it configures as the casafan, instead of ovlaim? Does it use the product id or is there another way that I don't seem to see? 
In any case, it does work fine after copying your current ovlaim dp17 code into the casafan file so there's no rush for a fix.
```
- id: 17
        name: color_temp
        type: integer
        range:
          min: 0
          max: 100
        mapping:
          - target_range:
              min: 3000
              max: 6000
```

**Reply by make-all** [reply, marked as answer]
Device detection currently uses matching of dps ids and types. It is not possible to query the product_id from devices after connection as far as I know, so the product_id is only there for future use in device discovery using broadcast messages (where it does appear).

**Reply by lk229** [reply]
Thanks. I downloaded your modified device file and removed and reinstalled my ceiling fans. I now have the option in the pull-down list of the correct device and everything works.
